packages/application/src/connectionlost.ts

```diff
@@ -22,6 +25,45 @@ export const ConnectionLost: IConnectionLost = async function (
       'JupyterLab will continue trying to reconnect.\n' +
       'Check your network connection or Jupyter server configuration.\n'
   );
+  const buttons = [Dialog.okButton({ label: trans.__('Dismiss') })];
+
+  if (!displayConnectionLost) {
+    return;
+  }
+
+  const key = title + '----' + networkMsg;
```

Comment:
```suggestion
  const key = 'server-connection-error';
```

Localized strings aren't typically used for internal record-keeping like this, due to the admittedly small risk that two strings in English could translate to the same string in some other language. I think it's fine to use a non-translated string in this case.